sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -781,3 +782,65 @@ case class SchemaOfJson(
 
   override def prettyName: String = "schema_of_json"
 }
+
+/**
+ * A function that returns number of elements in outer Json Array.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(jsonArray) - Returns length of the jsonArray",
+  arguments = """
+    jsonArray - A JSON array is required as argument. `Analysis Exception` is thrown if any other
+    valid JSON expression is passed. `NULL` is returned in case of invalid JSON.
+  """,
+  examples = """
+    Examples:
+    > SELECT _FUNC_('[1,2,3,4]');
+      4
+    > SELECT _FUNC_('[1,2,3,{"f1":1,"f2":[5,6]},4]');
+      5
+    > SELECT _FUNC_('[1,2');
+      NULL
+  """,
+  since = "3.1.0"
+)
+case class LengthOfJsonArray(child: Expression)
+  extends UnaryExpression with CodegenFallback {
+  override def dataType: DataType = IntegerType
+  override def nullable: Boolean = true
+  override def prettyName: String = "json_array_length"
+
+  override def eval(input: InternalRow): Any = {
+    @transient
+    val json = child.eval(input).asInstanceOf[UTF8String]
+    try {
+      Utils.tryWithResource(CreateJacksonParser.utf8String(SharedFactory.jsonFactory, json)) {
+        parser => {
+          // return null if null array is encountered.
+          if (parser.nextToken() == null) {
+            return null
+          }
+          // Parse the array to compute its length.
+          parseCounter(parser, input)
+        }
+      }
+    } catch {
+      case _: JsonProcessingException => null
+    }
+  }
+
+  private def parseCounter(parser: JsonParser, input: InternalRow): Int = {
+    // Counter for length of array
+    var array_length: Int = 0;
+    // Only json array are supported for this function.
+    if (parser.getCurrentToken != JsonToken.START_ARRAY) {
+      throw new AnalysisException(s"$prettyName can only be called on Json Array.")
+    }
+    // Keep traversing until the end of Json Array
+    while(parser.nextToken() != JsonToken.END_ARRAY) {
```

Comment:
Can nextToken return null? Looks like it can:
https://github.com/apache/spark/blob/8e280cebf25e47bf40df224461a76fc4c84cc997/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala#L28-L33

Reply:
It returns null when end of input is reached.
If it returns null before returning `END_ARRAY` then our json is invalid. Invalid input was already handled.
Anyway now i will add one more check for null.